sources/packages/backend/apps/api/src/route-controllers/note/_tests_/e2e/note.institutions.controller.getStudentNotes.e2e-spec.ts

```diff
@@ -0,0 +1,238 @@
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    //Act/Assert
+    return request(app.getHttpServer())
+      .get("/institutions/note/student/999999?noteType=invalid_node_type")
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.BAD_REQUEST)
+      .expect({
+        statusCode: 400,
+        message: "Validation failed (enum string is expected)",
+        error: "Bad Request",
+      });
+  });
+
+  it("Should get an empty student notes result when student has no notes.", async () => {
+    // Arrange
+    const savedApplication = await saveFakeApplication(db.dataSource, {
+      institutionLocation: collegeFLocation,
+    });
+    const student = savedApplication.student;
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    // Act/Assert
+    return request(app.getHttpServer())
+      .get(`/institutions/note/student/${student.id}`)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({});
+  });
+
+  it("Should get specific student notes types when student has notes and a note type filter was provided.", async () => {
+    // Arrange
+    const savedApplication = await saveFakeApplication(db.dataSource, {
+      institutionLocation: collegeFLocation,
+    });
+
+    const student = savedApplication.student;
+    const user = await db.user.save(createFakeUser());
+    const expectedNote = createFakeNote(NoteType.Application);
+    await saveFakeStudentNotes(
+      db.dataSource,
+      [
+        createFakeNote(NoteType.General),
+        expectedNote,
+        createFakeNote(NoteType.Designation),
+      ],
+      student.id,
+      user,
+    );
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    // Act/Assert
+    return request(app.getHttpServer())
+      .get(
+        `/institutions/note/student/${student.id}?noteType=${NoteType.Application}`,
+      )
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect([noteToApiReturn(expectedNote)]);
+  });
+
+  it("Should not get notes for student restriction with notification type 'no effect' when any are available.", async () => {
+    // Arrange
+    const savedApplication = await saveFakeApplication(db.dataSource, {
+      institutionLocation: collegeFLocation,
+    });
+    const student = savedApplication.student;
+    // Get any restriction with notification type "No effect".
+    const notificationTypeNoEffectRestriction = await db.restriction.findOne({
+      where: {
+        notificationType: RestrictionNotificationType.NoEffect,
+      },
+    });
+    await saveFakeStudentRestriction(db.dataSource, {
+      student,
+      restriction: notificationTypeNoEffectRestriction,
+      application: savedApplication,
+    });
+    const user = await db.user.save(createFakeUser());
```

Comment:
Unless the user is needed at the E2E scope I would leave its creation to `saveFakeStudentNotes`.

Reply:
I had the same opinion https://github.com/bcgov/SIMS/pull/1933/files#r1190434379.